app/controllers/idv_controller.rb

```diff
@@ -20,6 +21,7 @@ def index
       )
       redirect_to idv_session_errors_failure_url
     else
+      clear_session if enrollment_status == 'expired'
```

Comment:
Since this logic is shared for remote supervised as well, it might be worth qualifying the enrollment here as being related to in-person. Alternatively, I wonder if the link in the email could direct the user to an in-person-specific route (like `InPersonController#index`), where we wouldn't have to differentiate, and where we wouldn't need to have extra database queries happening at the start of every IdV session.

Reply:
interesting point. for the 1st option - we are getting the status from an `InPersonEnrollment` on line 37; are you suggesting we add a check for `in_person_proofing_enabled` in addition to the status? 
For opt 2 I guess I wonder a) if design has strong opinions on where the user is routed to from the `start over` link in the email and b) do we expect users who have an expired enrollment to only be able to proof in person if they were to start the process again? Or would we want to allow them to be able to take the remote path when they start verification again?

Reply:
But also maybe since it's a bug we're much more likely to encounter rather than actual users are my thoughts around opt. 2 relevant?

Reply:
i tried moving this logic to that block you linked to but what happens is the user can no longer progress past the welcome step. i'm not sure why that is happening?

Reply:
The more I think about this, the more I think there's an issue in how we consider the user's `pending_in_person_enrollment`. It seems like a user shouldn't be allowed to get into a state where they have an expired profile which is newer than a profile which is considered pending, which seems like it might be technically possible based on [how those are implemented](https://github.com/18F/identity-idp/blob/38c8c65ee01a6dcd72f6e2a64cd7460a78e711cf/app/models/user.rb#L48-L56) and would be the only way I could imagine the barcode screen would be shown when we're not expecting it to be.

If we were able to update the logic so that `pending_in_person_enrollment` would return `nil` if there's a more recent enrollment, then I expect that could potentially fix the problem?

Reply:
I don't know if the issue is a pending status. It looks like the session still has the applicant info and so is redirecting to `final_url` https://github.com/18F/identity-idp/blob/bd5d12b9becabc663546f96d929461dc13af2d68/app/controllers/idv/doc_auth_controller.rb#L40